src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/StringParsingHelpers.Connections.cs

```diff
@@ -99,15 +89,10 @@ internal static TcpConnectionInformation[] ParseActiveTcpConnectionsFromFiles(st
 
         internal static IPEndPoint[] ParseActiveTcpListenersFromFiles(string tcp4ConnectionsFile, string tcp6ConnectionsFile)
         {
-            if (!File.Exists(tcp4ConnectionsFile) || !File.Exists(tcp6ConnectionsFile))
-            {
-                throw new PlatformNotSupportedException(SR.net_InformationUnavailableOnPlatform);
-            }
-
-            string tcp4FileContents = File.ReadAllText(tcp4ConnectionsFile);
+            string tcp4FileContents = ReadAllText(tcp4ConnectionsFile);
```

Comment:
Why are we getting rid of the file existence checks?

Reply:
The reason was to change the type of exception thrown to something more appropriate. PNSE is not mentioned by the docs and feels more appropriate for cases when there is no (implemented?) way of accessing the information in question.

Omitting the file existence checks will still make the code throw, but it will throw `NetworkInformationException` instead (which is mentioned in the docs), and its inner exception will provide more info to the user.